src/core/plugin/luapi_application.h

```diff
@@ -2280,6 +2280,35 @@ static int applib_scrollToPos(lua_State* L) {
     return 0;
 }
 
+/**
+ * Obtains the label of the specified page in the pdf background.
+ *
+ * @param page integer
+ * @return string|nil Returns the label on success and (nil, message) if page number is out of range.
+ * @return string
+ *
+ * Example 1: local label = app.getPageLabel(10)
+ ' obtains the page label of page 10 in the background pdf
+ **/
+static int applib_getPageLabel(lua_State* L) {
+    const Plugin* plugin = Plugin::getPluginFromLua(L);
+    const Control* control = plugin->getControl();
+    const XojPdfDocument& doc = control->getDocument()->getPdfDocument();
+
+    const size_t pagenr = static_cast<size_t>(luaL_checkinteger(L, 1)) - 1;
+    const size_t last = doc.getPageCount() - 1;
+
+    if (pagenr > last) {
```

Comment:
This looks safer (just for case there is no pdf page)
```suggestion

    if (pagenr >= doc.getPageCount()) {
```